src/main/js/main_html5.js

```diff
@@ -455,7 +455,24 @@ require("../../../html5-common/js/utils/environment.js");
       if (_video.seeking) {
         playQueued = true;
       } else {
-        executePlay(false);
+        var playPromise = executePlay(false);
+        if (playPromise) {
+          if (typeof playPromise.catch === 'function') {
```

Comment:
Do we need to check for both `catch` and `then`? I think the code might be a bit cleaner with a single check and the browser wouldn't support one without the other. Not super important though

Reply:
Will polyfills support both? I'm hoping yes, but we can't be sure. What do you think? I can remove the checks.

Reply:
@aeng7 since the promise is returned by the video element I don't think polyfills would come into play. I guess it's possible to have a bad polyfill that overrides `window.Promise` even if it exists, but it would be kind of an edge case. We can probably keep it like this to be safe though

Reply:
Yea I'd rather not include these checks as well to be cleaner, but I noticed that we were checking existence previously:

https://github.com/ooyala/html5-video-plugins/blob/a702f36aec6df2cfcc438d5d9f9c3da5628fed14/src/main/js/main_html5.js#L583

We can probably remove all these checks at a safer time.

Reply:
Ok sounds good